utils/symbol.c

```diff
@@ -1430,3 +1423,71 @@ void print_symtabs(struct symtabs *symtabs)
 		symbol_putname(sym, name);
 	}
 }
+
+static unsigned long kaddr;
+
+static unsigned long get_kaddr(char *str)
+{
+	char *pch;
+	unsigned long addr;
+
+	pch = strtok (str,"-");
+	addr = strtoul(pch, NULL, 16);
+
+	if (addr < 0x40000000UL) {
+		return 0x40000000UL;
+	} else if (addr < 0x80000000UL) {
+		return 0x80000000UL;
+	} else if (addr < 0xB0000000UL) {
+		return 0xB0000000UL;
+	} else if (addr < 0xC0000000UL) {
+		return 0xC0000000UL;
+#if __SIZEOF_LONG__ == 8
+	} else {
+		return 0x800000000000ULL;
+	}
+#else
+    }
+#endif
+}
+
+void set_kaddr(char *dirname, const char *session_id)
+{
+	FILE *fp;
+	char buf[4096];
+	char line[200];
+
+	snprintf(buf, sizeof(buf), "%s/sid-%s.map", dirname, session_id);
+
+	fp = fopen(buf, "r");
+
+	if (fp == NULL)
+		exit(1);
```

Comment:
I suggest to remove a blank line between `fopen()` and the if statement.  Also please you `pr_err()` instead of the plain `exit()` so that it can show the error message on failure.

In addition, you can share a single buffer and get rid of `line`.